ui/ui-frontend-common/src/app/modules/object-editor/services/edit-object.service.spec.ts

```diff
@@ -0,0 +1,671 @@
+    ));
+
+    it('should create with empty data', waitForAsync(
+      inject(
+        [MockSchemaService, TemplateService, SchemaService],
+        (schemaService: MockSchemaService, templetaService: TemplateService, ss: SchemaService) => {
+          schemaService.getSchema(Collection.ARCHIVE_UNIT).subscribe((schema) => {
+            const path = '';
+            const originalData = {};
+            const template: DisplayRule[] = [];
+            const projectedData = templetaService.toProjected(originalData, template);
+            const subschema = schema.filter((element) => element.Category === 'DESCRIPTION');
+            const templatedSchema = service.createaTemplateSchema(template, subschema);
+            const editObject: EditObject = service.editObject(path, projectedData, template, templatedSchema);
+
+            expect(editObject).toBeTruthy();
+            expect(editObject.children.length).toEqual(16);
+            expect(editObject.children).toEqual(
+              jasmine.arrayContaining(
+                [
+                  'Addressee',
+                  'Agent',
+                  'AuthorizedAgent',
+                  'Coverage',
+                  'DescriptionLevel',
+                  'Event',
+                  'Gps',
+                  'Recipient',
+                  'Sender',
+                  'Signature',
+                  'SigningInformation',
+                  'Title',
+                  'Transmitter',
+                  'Writer',
+                  '#originating_agency',
+                  '#originating_agencies',
+                ].map((path) => jasmine.objectContaining({ path })),
+              ),
+            );
+            expect(editObject.children).toEqual(
+              jasmine.arrayContaining([
+                jasmine.objectContaining({
+                  path: 'Addressee',
+                  kind: 'object-array',
+                  children: [],
+                }),
+              ]),
+            );
+          });
+        },
+      ),
+    ));
+  });
+
+  describe('Kind', () => {
+    it('should object be object kind', () => {
+      expect(service.kind({})).toEqual('object');
+    });
+
+    // TODO: Determine which kind is empty array
+    xit('should empty array be object-array kind', () => {
+      expect(service.kind([])).toEqual('object-array');
+    });
+
+    it('should string array be primitive-array kind', () => {
+      expect(service.kind(['a', 'b', 'c'])).toEqual('primitive-array');
+    });
+
+    it('should object array be object-array kind', () => {
+      expect(service.kind([{}, {}, {}])).toEqual('object-array');
+    });
+
+    it('should null be object kind', () => {
+      expect(service.kind(null)).toEqual('object');
+    });
+  });
+
+  describe('TemplateSchema', () => {
+    it('should create template schema without template and schema', () => {
+      const template = [];
+      const schema = [];
+      const ouput = service.createaTemplateSchema(template, schema);
```

Comment:
```suggestion
      const output = service.createaTemplateSchema(template, schema);
```
À corriger sur toutes les occurrences de `ouput`